### ISSUE TITLE
[DragonAPI] Changing "Disallow Null-Item ItemStacks to Prevent Crashes" appears to do nothing

### ISSUE THREAD

**Issue by Kachajal** [ReikaKalseki/Reika_Mods_Issues#1214]
Using DragonAPI 1.7.10 v16D. Minecraft 1.7.10 with Forge 10.13.4.1614-1.7.10.

Full error log is [here](http://pastebin.com/JXQxQ6sG).

I'm trying to use Tome of Alkahestry from Reliquary 1.2.257. Unfortunately, it seems to have some null stack output recipes. DragonAPI takes exception to that and removes them to prevent crashing, as stated in the log:

    [15:52:29] [Client thread/INFO]: DRAGONAPI: Invalid recipe, such as with nulled inputs, found. Removing to prevent crashes. Class=class xreliquary.items.alkahestry.AlkahestryDrainRecipe, Output is a null-item ItemStack.
    [15:52:29] [Client thread/INFO]: DRAGONAPI: Invalid recipe, such as with nulled inputs, found. Removing to prevent crashes. Class=class xreliquary.items.alkahestry.AlkahestryRedstoneRecipe, Output is a null-item ItemStack.
    [15:52:29] [Client thread/INFO]: DRAGONAPI: Invalid recipe, such as with nulled inputs, found. Removing to prevent crashes. Class=class xreliquary.items.alkahestry.AlkahestryCraftingRecipe, Output is a null-item ItemStack.

Which is all fine and dandy, but seeing as the recipes don't seem to actually cause crashing, I tried changing the 

        B:"Disallow Null-Item ItemStacks to Prevent Crashes"=true

option to false, but it didn't seem to have any effect. The log was exactly the same as far as I could see, and the recipes were still disabled. 

Am I doing something wrong, or is this a bug?

Thanks in advance!

**Comment by ReikaKalseki** [comment]
It was never fully implemented, and that recipe removal is not configurable.

**Comment by ReikaKalseki** [comment]
This might count as the oldest reopening of an issue ever.

**Comment by EricPfau** [comment]
I appreciate you looking into this given how old it is. If there is nothing you can do about it, it's still great that you even looked. Thanks Reika.

**Comment by ReikaKalseki** [comment]
OK, thinking about it a little bit, I remember this now.

That option (~2014 or 2015) was originally intended to outright disallow any construction of an ItemStack of null. I promptly realized the problem with that plan when I noticed vanilla does it in several locations (donkey chests being the example I remember).

I then abandoned the idea, but had already coded in the option. That recipe filter system is far newer (~2017) and was piggybacking on a recipe handler system I added for my own modpack (as it was far more performant to just modify/cancel recipes on addition versus iterating the recipe list repeatedly for each kind of change). It was never meant to work with that option.

There is no reason to want recipes with null items in them - they WILL crash in many handlers. As such, I have removed that config entry.
